packages/wallet-apis/src/provider.ts

```diff
@@ -37,30 +40,29 @@ export type SmartWalletClient1193Methods = [
 
 const handler = createEip1193HandlerFactory<SmartWalletClient1193Methods>();
 
-export const createEip1193ProviderFromClient = <
-  TAccount extends Address | undefined = Address | undefined,
->(
-  client: BaseWalletClient<SmartWalletActions<TAccount>>,
+export const createEip1193Provider = (
+  clientParams: CreateSmartWalletClientParams<undefined>,
+  accountOptions: CreationOptions | { accountAddress: Address } = {},
 ) => {
+  let client: SmartWalletClient<Address> | undefined;
+
   // TODO(v5): implement any other supported events: https://eips.ethereum.org/EIPS/eip-1193#events
   const eventEmitter = new EventEmitter();
 
-  // TODO(v5): revisit if we actually want to auto-connect the provider like this.
-  (async () => {
-    const account = await client.requestAccount();
-    client.account = {
-      type: "json-rpc",
-      address: account.address,
-    };
-    eventEmitter.emit("connect", { chainId: toHex(client.chain.id) });
-  })();
-
   const request = (async ({ method, params }) => {
+    client ??= await (async () => {
```

Comment:
Does this work? `client` doesn't get set until the `await` completes, so I think multiple requests will still all see the client as absent and try to initialize it.

Reply:
Gonna merge now so I can try out the new provider in my PR. Will update this there.

Reply:
ah you're right

Reply:
thanks